phper/src/builder/argument.rs

```diff
@@ -0,0 +1,138 @@
+use crate::{
+    builder::{
+        data_type::AtomicType,
+        describe::Describe,
+        value::{ConstantValue, Value},
+    },
+    errors::{Error, Result},
+};
+
+pub enum ArgumentFlag {
+    IsVaradic,
+    HasDefault(ConstantValue),
+}
+
+pub struct Argument {
+    pub name: String,
+    pub data_type: AtomicType,
+    pub flag: Option<ArgumentFlag>,
+    pub value: Option<Value>,
+}
+
+impl Argument {
+    pub fn new(name: &str, data_type: AtomicType) -> Self {
+        Self {
+            name: name.to_string(),
+            data_type,
+            flag: None,
+            value: None,
+        }
+    }
+
+    pub fn varadic(name: &str, data_type: AtomicType) -> Self {
+        Self {
+            name: name.to_string(),
+            data_type,
+            flag: Some(ArgumentFlag::IsVaradic),
+            value: None,
+        }
+    }
+
+    pub fn with_default(mut self, value: ConstantValue) -> Self {
+        self.flag = Some(ArgumentFlag::HasDefault(value));
+
+        self
+    }
+
+    pub fn value(&self) -> Result<Value> {
+        if let Some(v) = &self.value {
+            Ok(v.clone())
+        } else if let Some(ArgumentFlag::HasDefault(default)) = &self.flag {
+            let value = default.clone().into();
+
+            Ok(value)
+        } else {
+            Err(Error::other(
+                "attempted to access argument value prior to initlization.",
+            ))
+        }
+    }
+}
+
+/// Describe `Argument`.
+///
+/// Example:
+///
+/// ```rust
+/// use phper::builder::{
+///     argument::{Argument, ArgumentFlag},
+///     data_type::AtomicType,
+///     describe::Describe,
+///     value::ConstantValue,
+/// };
```

Comment:
Add `, no_run` after the rust to solve the doc test problem.

Reply:
thanks! is there a way to make DOC tests run otherwise?